dds/idl/ts_generator.cpp

```diff
@@ -84,15 +84,36 @@ bool ts_generator::generate_ts(AST_Decl* node, UTL_ScopedName* name)
     IDL_GlobalData::DCPS_Data_Type_Info* info = idl_global->is_dcps_type(name);
     if (info) {
       key_count = info->key_list_.size();
-    } else if (be_global->is_topic_type(struct_node)) {
+    } else if (be_global->treat_as_topic(struct_node)) {
+
+      ///Go ahead and warn if we see bad stuff
+      be_global->is_nested_type(node);
+
       key_count = TopicKeys(struct_node).count();
+    } else if(be_global->is_nested_type(node)) {
+      ///Being explicit to prevent any confusion and allow for modification in
+      //the future.
+
+      /**
+       * @TODO: only run is_nested_type(node) and treat_as_topic(*_node) once
+       */
+      return true;
+
     } else {
       return true;
     }
-  } else if (be_global->is_topic_type(union_node)) {
+  } else if (be_global->treat_as_topic(union_node)) {
+
+    ///Go ahead and warn if we see bad stuff
+    be_global->is_nested_type(node);
+
     key_count = be_global->has_key(union_node) ? 1 : 0;
   } else {
-    return true;
+    /**
+     * @NOTE: this falls through if be_global->is_nested_type(node) == true)
+     * @TODO: only run is_nested_type(node) and treat_as_topic(*_node) once
+     */
```

Comment:
You can have it check just once (and always) by moving the warning logic to `dds_generator.cpp`.

Reply:
Sorry, it's `dds_visitor.cpp` is where the check should happen. You could do it from `dds_generator.cpp` as well, but it's better to do it from visitor because that's where other checks are being done.